distributed/s3.py

```diff
@@ -486,58 +486,76 @@ def read_block_from_s3(filename, offset, length, s3pars={}, delimiter=None):
 
 
 @gen.coroutine
-def _read_text(fn, encoding='utf-8', errors='strict', lineterminator='\n',
-               executor=None, fs=None, lazy=True, collection=True):
-    from dask import do
+def _read_text(fn, keyname=None, encoding='utf-8', errors='strict', lineterminator='\n',
```

Comment:
It's idiomatic in S3 to refer to buckets and keynames separately.

What actually drove me to do this though was backwards compatibility with the previous version, as described in an [early blogpost](http://matthewrocklin.com/blog/work/2016/02/17/dask-distributed-part1)